RUMM-397 RUM Documentation (#238)
DataDog/dd-sdk-android: docs/rum_collection.md

> # Android Trace Collection

Reviewer: ```suggestion
# Android RUM Collection
```

> <div class="alert alert-info">The Android trace collection is in public beta, and is currently not supported by Datadog.</div>

Reviewer: ```suggestion
<div class="alert alert-info">The Android rum collection is in public beta, and is currently not supported by Datadog.</div>
```

Reviewer 2: Remember to properly capitalize RUM

    }

    dependencies {
        implementation "com.datadoghq:dd-sdk-android:x.x.x"
    }
    ```

2. Initialize the library with your application context and your [Datadog client token][4]. For security reasons, you must use a client token: you cannot use [Datadog API keys][5] to configure the `dd-sdk-android` library as they would be exposed client-side in the Android application APK byte code. For more information about setting up a client token, see the [client token documentation][4]. You also need to provide an Application ID (see our [RUM Getting Started page][7]).

    {{< tabs >}}
    {{% tab "US" %}}

```kotlin
class SampleApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        val config = DatadogConfig.Builder(BuildConfig.DD_CLIENT_TOKEN, BuildConfig.APPLICATION_ID)
                        .trackGestures()
                        .useViewTrackingStrategy(strategy)
                        .build()
        Datadog.initialize(this, config)
    }
}
```

    {{% /tab %}}
    {{% tab "EU" %}}

```kotlin
class SampleApplication : Application() {
    override fun onCreate() {
        super.onCreate()

>         val config = DatadogConfig.Builder(BuildConfig.DD_CLIENT_TOKEN, BuildConfig.APPLICATION_ID)

Reviewer: now that we are making the "environment" mandatory I think we should add it here already. Please check my PR later today.

                                             .trackGestures()
                                             .useViewTrackingStrategy(strategy)
                        .useEUEndpoints()
                        .build()
        Datadog.initialize(this, config)
    }
}
```

    {{% /tab %}}
    {{< /tabs >}}
    

> Depending on your application's architecture, you can choose one of several implementations of ViewTrackingStrategy :

Reviewer 2: ```suggestion
Depending on your application's architecture, you can choose one of several implementations of ViewTrackingStrategy:
```

>  - ActivityViewTrackingStrategy: every Activity in your application is considered a distinct View;

Reviewer 2: ```suggestion
 - `ActivityViewTrackingStrategy`: Every activity in your application is considered a distinct view.
```

>  - FragmentViewTrackingStrategy: every Fragment in your application is considered a distinct View;

Reviewer 2: ```suggestion
 - `FragmentViewTrackingStrategy`: Every fragment in your application is considered a distinct view.
```

>  - NavigationViewTrackingStrategy: if you use the Android Jetpack Navigation library, this is the recommended strategy. It will automatically track Navigation destination as a distinct View;

Reviewer: should we mention also the fact that you can exclude some fragments/activities from tracking ?

Author: This documentation is a general getting started I guess. The javadoc when they start using it will tell them.

>  - NavigationViewTrackingStrategy: if you use the Android Jetpack Navigation library, this is the recommended strategy. It will automatically track Navigation destination as a distinct View;

Reviewer 2: ```suggestion
 - `NavigationViewTrackingStrategy`: If you use the Android Jetpack Navigation library, this is the recommended strategy. It automatically tracks the navigation destination as a distinct view.
```

> 3. Configure and register the Rum Monitor. You only need to do it once, usually in your application's `onCreate()` method:

Reviewer 2: ```suggestion
3. Configure and register the RUM Monitor. You only need to do it once, usually in your application's `onCreate()` method:
```

    ```kotlin
    val monitor = RumMonitor.Builder().build()
    GlobalRum.registerIfAbsent(monitor)
    ```
   

> 4. If you want to track your OkHttp requests as resources, you can add the provided [Interceptor][6] as follow:

Reviewer 2: ```suggestion
4. If you want to track your OkHttp requests as resources, you can add the provided [Interceptor][6] as follows:
```

    ```kotlin
    val okHttpClient =  OkHttpClient.Builder()

>         .addInterceptor(DatadogRumMonitor())

Reviewer: ```suggestion
        .addInterceptor(RumInterceptor())
```

Author: The DatadogInterceptor is recomended as it will automatically bundle RUM and Traces in the same interceptor.

Reviewer: But you were having `DatadogRumMonitor` there not `DatadogInterceptor`

Author: Oups

        .build()
    ```

>     This will create a RUM resource data around each request processed by the OkHttpClient, with all the relevant information automatically filled (url, method, status code, error).

Reviewer 2: ```suggestion
    This creates RUM resource data around each request processed by the OkHttpClient, with all the relevant information automatically filled (URL, method, status code, error).
```

>     **Note**: If you use multiple Interceptors, this one must be called first.  

Reviewer: If think we can avoid this by mentioning them the "CombinedInterceptor" option

Author: DatadogInterceptor is the Combined interceptor
